core/Command/Maintenance/Repair.php

```diff
@@ -102,47 +109,36 @@ protected function execute(InputInterface $input, OutputInterface $output): int
-		$this->dispatcher->addListener('\OC\Repair::advance', [$this, 'handleRepairFeedBack']);
-		$this->dispatcher->addListener('\OC\Repair::finishProgress', [$this, 'handleRepairFeedBack']);
-		$this->dispatcher->addListener('\OC\Repair::step', [$this, 'handleRepairFeedBack']);
-		$this->dispatcher->addListener('\OC\Repair::info', [$this, 'handleRepairFeedBack']);
-		$this->dispatcher->addListener('\OC\Repair::warning', [$this, 'handleRepairFeedBack']);
-		$this->dispatcher->addListener('\OC\Repair::error', [$this, 'handleRepairFeedBack']);
+		$this->dispatcher->addListener(RepairStartEvent::class, [$this, 'handleRepairFeedBack']);
+		$this->dispatcher->addListener(RepairAdvanceEvent::class, [$this, 'handleRepairFeedBack']);
+		$this->dispatcher->addListener(RepairFinishEvent::class, [$this, 'handleRepairFeedBack']);
+		$this->dispatcher->addListener(RepairStepEvent::class, [$this, 'handleRepairFeedBack']);
+		$this->dispatcher->addListener(RepairInfoEvent::class, [$this, 'handleRepairFeedBack']);
+		$this->dispatcher->addListener(RepairWarningEvent::class, [$this, 'handleRepairFeedBack']);
+		$this->dispatcher->addListener(RepairErrorEvent::class, [$this, 'handleRepairFeedBack']);
 
 		$this->repair->run();
 
 		$this->config->setSystemValue('maintenance', $maintenanceMode);
 		return 0;
 	}
 
-	public function handleRepairFeedBack($event) {
-		if (!$event instanceof GenericEvent) {
-			return;
-		}
-		switch ($event->getSubject()) {
-			case '\OC\Repair::startProgress':
-				$this->progress->start($event->getArgument(0));
-				break;
-			case '\OC\Repair::advance':
-				$this->progress->advance($event->getArgument(0));
-				break;
-			case '\OC\Repair::finishProgress':
-				$this->progress->finish();
-				$this->output->writeln('');
-				break;
-			case '\OC\Repair::step':
-				$this->output->writeln(' - ' . $event->getArgument(0));
-				break;
-			case '\OC\Repair::info':
-				$this->output->writeln('     - ' . $event->getArgument(0));
-				break;
-			case '\OC\Repair::warning':
-				$this->output->writeln('     - WARNING: ' . $event->getArgument(0));
-				break;
-			case '\OC\Repair::error':
-				$this->output->writeln('<error>     - ERROR: ' . $event->getArgument(0) . '</error>');
-				break;
+	public function handleRepairFeedBack(Event $event): void {
+		if ($event instanceof RepairStartEvent) {
+			$this->progress->start($event->getMaxStep());
+		} elseif ($event instanceof RepairAdvanceEvent) {
+			$this->progress->advance($event->getCurrentStep());
+		} elseif ($event instanceof RepairFinishEvent) {
+			$this->progress->finish();
+			$this->output->writeln('');
+		} elseif ($event instanceof RepairStepEvent) {
+			$this->output->writeln(' - ' . $event->getStepName());
+		} elseif ($event instanceof RepairInfoEvent) {
+			$this->output->writeln('     - ' . $event->getMessage());
+		} elseif ($event instanceof RepairWarningEvent) {
+			$this->output->writeln('     - WARNING: ' . $event->getMessage());
```

Comment:
While we are on it:
```suggestion
			$this->output->writeln('<comment>     - WARNING: ' . $event->getMessage() . '</comment>');
```

For yellow font on default background?

Reply:
Only for the warning line or all of them?

Reply:
warning as that is then yellow like the error is then red.

Reply:
https://github.com/nextcloud/server/pull/33640/commits/652ab6da3f5309b0ea6e1e19ce4d915ecb6cc9b1 ?

Reply:
We will see if that is too color full 👯